src/cursor/cur_table.c

```diff
@@ -837,18 +868,29 @@ __curtable_largest_key(WT_CURSOR *cursor)
 static int
 __curtable_bound(WT_CURSOR *cursor, const char *config)
 {
-    WT_CURSOR *primary;
+    WT_CURSOR **cp, *primary;
+    WT_CURSOR_BOUNDS_STATE saved_bounds;
     WT_CURSOR_TABLE *ctable;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
+    u_int i, j;
 
+    i = 0;
     ctable = (WT_CURSOR_TABLE *)cursor;
+    primary = *ctable->cg_cursors;
     JOINABLE_CURSOR_API_CALL(cursor, session, bound, NULL);
 
-    /* Grab the primary cursor and call bound function. */
-    primary = *ctable->cg_cursors;
-    WT_ERR(primary->bound(primary, config));
+    __wt_cursor_bounds_save(primary, &saved_bounds);
+
+    /* Call bound function on all column groups. */
+    for (i = 0, cp = ctable->cg_cursors; i < WT_COLGROUPS(ctable->table); i++, cp++)
+        WT_ERR((*cp)->bound(*cp, config));
 err:
+    /* If applying bounds fails on one colgroup, reset all of them for consistency. */
+    if (ret != 0) {
+        for (j = 0, cp = ctable->cg_cursors; j <= i; j++, cp++)
+            (__wt_cursor_bounds_restore(*cp, &saved_bounds));
```

Comment:
Why does this have brackets?

Reply:
[59a73d6](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/59a73d6177d493f37cd1d9b9e8a5f5b0b9edb277)